pkg/v21/chartconfig/current.go

```diff
@@ -12,21 +12,22 @@ import (
 
 	"github.com/giantswarm/cluster-operator/pkg/label"
 	"github.com/giantswarm/cluster-operator/pkg/project"
+	"github.com/giantswarm/cluster-operator/pkg/v21/controllercontext"
 )
 
 func (c *ChartConfig) GetCurrentState(ctx context.Context, clusterConfig ClusterConfig) ([]*v1alpha1.ChartConfig, error) {
-	c.logger.LogCtx(ctx, "level", "debug", "message", "looking for chartconfigs in the tenant cluster")
-
-	tenantG8sClient, err := c.newTenantG8sClient(ctx, clusterConfig)
+	cc, err := controllercontext.FromContext(ctx)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
+	c.logger.LogCtx(ctx, "level", "debug", "message", "looking for chartconfigs in the tenant cluster")
+
 	listOptions := metav1.ListOptions{
 		LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
 	}
 
-	chartConfigList, err := tenantG8sClient.CoreV1alpha1().ChartConfigs(resourceNamespace).List(listOptions)
+	chartConfigList, err := cc.Client.TenantCluster.G8s.CoreV1alpha1().ChartConfigs(resourceNamespace).List(listOptions)
```

Comment:
It might happen that the clients are not yet available. In such cases the resource should be canceled. Otherwise it would panic. Let's make sure we have the checks in in all resources using the clients from the context.

Reply:
Ack. Good point I'll add that.

Reply:
Done. I'll do the migration resources in my next PR.